site/utils.js

```diff
@@ -38,100 +38,136 @@ const stores = {
-    }
-    return null;
+  const url = new URL(window.location.href);
+  return url.searchParams.get(name);
 }
 
+/**
+ * @description Converts a string to a number
+ * @param {string} value String to convert
+ * @param {number} defaultValue Default value if conversion fails
+ * @returns {number} Converted number or default value
+ */
 function toNumber(value, defaultValue) {
-    try {
-        return Number.parseFloat(value);
-    } catch (e) {
-        return defaultValue;
-    }
+  try {
+    return Number.parseFloat(value);
+  } catch (e) {
+    return defaultValue;
+  }
 }
 
+/**
+ * @description Create dom element from html string and add inner html via string template
+ * @param {string} el Element type
+ * @param {string} html Inner html
+ * @returns {HTMLElement} DOM element
+ */
 function dom(el, html) {
-    let element = document.createElement(el);
-    element.innerHTML = html;
-    return element;
+  const element = document.createElement(el);
+  element.innerHTML = html;
+  return element;
 }
 
 async function loadItems() {
-    const response = await fetch("latest-canonical.json")
-    const items = await response.json();
-
-    for (item of items) {
-        item.search = item.name + " " + item.unit;
-        item.search = item.search.toLowerCase().replace(",", ".");
-
-        item.numPrices = item.priceHistory.length;
-        item.priceOldest = item.priceHistory[item.priceHistory.length - 1].price;
-        item.dateOldest = item.priceHistory[item.priceHistory.length - 1].date;
-        item.date = item.priceHistory[0].date;
-        let highestPriceBefore = -1;
-        for (let i = 1; i < item.priceHistory.length; i++) {
-            const price = item.priceHistory[i];
-            highestPriceBefore = Math.max(highestPriceBefore, price.price);
-        }
-        if (highestPriceBefore == -1) highestPriceBefore = item.price;
-        item.highestBefore = highestPriceBefore;
+  const response = await fetch("latest-canonical.json");
+  const items = await response.json();
+
+  for (const item of items) {
+    item.search = item.name + " " + item.unit;
+    item.search = item.search.toLowerCase().replace(",", ".");
+
+    item.numPrices = item.priceHistory.length;
+    item.priceOldest = item.priceHistory[item.priceHistory.length - 1].price;
+    item.dateOldest = item.priceHistory[item.priceHistory.length - 1].date;
+    item.date = item.priceHistory[0].date;
+    let highestPriceBefore = -1;
+    for (let i = 1; i < item.priceHistory.length; i++) {
+      const price = item.priceHistory[i];
+      highestPriceBefore = Math.max(highestPriceBefore, price.price);
     }
-    return items;
+    if (highestPriceBefore == -1) highestPriceBefore = item.price;
+    item.highestBefore = highestPriceBefore;
+  }
+  return items;
 }
 
-let carts = [];
-loadCarts();
-
-function loadCarts() {
+/**
+ * @description Class for managing the shopping cart, which is stored in local storage
```

Comment:
It manages multiple carts, please use the plural form for the class name and docs.

Reply:
changed `jsdoc`